packages/util/helper.go

```diff
@@ -6,20 +6,238 @@ import (
 	"encoding/base64"
 	"encoding/hex"
 	"fmt"
+	"io"
 	"math/rand"
+	"net"
 	"os"
 	"os/exec"
 	"path"
 	"sort"
 	"strings"
+	"sync"
 	"time"
 
+	"encoding/json"
+	"net/http"
+	"path/filepath"
+
 	"github.com/Infisical/infisical-merge/packages/api"
 	"github.com/Infisical/infisical-merge/packages/models"
 	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
 )
 
+const (
+	RELAY_CACHE_FILE = "/var/lib/infisical/cached_relay"
+)
+
+type Relay struct {
+	Name      string    `json:"name"`
+	Host      string    `json:"host"`
+	Heartbeat time.Time `json:"heartbeat"`
+	OrgId     *string   `json:"orgId"`
+}
+
+func GetRelayName(cmd *cobra.Command, forceRefetch bool, accessToken string) (string, error) {
+	relayName, err := GetCmdFlagOrEnvWithDefaultValue(cmd, "relay", []string{"INFISICAL_RELAY_NAME"}, "")
+	if err != nil {
+		return "", fmt.Errorf("unable to parse relay flag: %v", err)
+	}
+	if relayName != "" {
+		return relayName, nil
+	}
+
+	domain, err := cmd.Flags().GetString("domain")
+	if err != nil {
+		return "", fmt.Errorf("unable to parse domain flag: %v", err)
+	}
+
+	relaysURL := fmt.Sprintf("%s/v1/relays", domain)
+
+	// fetch all relays from Infisical
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", relaysURL, nil)
+	if err != nil {
+		return "", fmt.Errorf("unable to create request to fetch relays: %v", err)
+	}
+
+	req.Header.Add("Authorization", "Bearer "+accessToken)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("unable to fetch relays: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		bodyBytes, _ := io.ReadAll(resp.Body)
+		return "", fmt.Errorf("unable to fetch relays: status code %d, body: %s", resp.StatusCode, string(bodyBytes))
+	}
```

Comment:
let's move this to the `api.go` file please.. and maybe follow the convention there for consistency

<img width="1095" height="685" alt="image" src="https://github.com/user-attachments/assets/7ead63cf-f8b5-443f-960f-9b9bb1166623" />